### ISSUE TITLE
compilation broken on main?

### ISSUE THREAD

**Issue by lucas-schuermann** [shux-lang/shux#118]
```
lucas@numel:~/shux/build$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
lucas@numel:~/shux/build$ make cleanall
rm -rf *.* shuxc
cd ../tests && rm -rf *.ll && rm -rf *.out
lucas@numel:~/shux/build$ make symlink
ln -s -f ../src/frontend/exceptions.ml exceptions.ml
ln -s -f ../src/frontend/ast.mli ast.ml
ln -s -f ../src/backend/sast.mli sast.ml
ln -s -f ../src/backend/cast.mli cast.ml
ln -s -f ../src/backend/codegen.ml codegen.ml
ln -s -f ../src/frontend/parser.mly parser.mly
ln -s -f ../src/frontend/scanner.mll scanner.mll
ln -s -f ../src/backend/semant.ml semant.ml
ln -s -f ../src/backend/sast_cast.ml sast_cast.ml
ln -s -f ../src/backend/ast_sast.ml ast_sast.ml
ln -s -f ../src/shuxc.ml shuxc.ml
ln -s -f ../src/frontend/astprint.ml astprint.ml
ln -s -f ../src/backend/llast.mli llast.ml
ln -s -f ../src/backend/lltranslate.ml lltranslate.ml
lucas@numel:~/shux/build$ make all
rm -rf *.* shuxc
cd ../tests && rm -rf *.ll && rm -rf *.out
ln -s -f ../src/frontend/exceptions.ml exceptions.ml
ln -s -f ../src/frontend/ast.mli ast.ml
ln -s -f ../src/backend/sast.mli sast.ml
ln -s -f ../src/backend/cast.mli cast.ml
ln -s -f ../src/backend/codegen.ml codegen.ml
ln -s -f ../src/frontend/parser.mly parser.mly
ln -s -f ../src/frontend/scanner.mll scanner.mll
ln -s -f ../src/backend/semant.ml semant.ml
ln -s -f ../src/backend/sast_cast.ml sast_cast.ml
ln -s -f ../src/backend/ast_sast.ml ast_sast.ml
ln -s -f ../src/shuxc.ml shuxc.ml
ln -s -f ../src/frontend/astprint.ml astprint.ml
ln -s -f ../src/backend/llast.mli llast.ml
ln -s -f ../src/backend/lltranslate.ml lltranslate.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c exceptions.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c sast.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c ast.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c astprint.ml
ocamllex scanner.mll
138 states, 6096 transitions, table size 25212 bytes
ocamlyacc parser.mly
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c parser.mli
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c scanner.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c parser.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c semant.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c codegen.ml
ocamlfind ocamlc -linkpkg -thread -package core -package llvm -package llvm.analysis -I ../src/include -g exceptions.cmo sast.cmo ast.cmo astprint.cmo scanner.cmo parser.cmo semant.cmo codegen.cmo ast_sast.cmo sast_cast.cmo llast.cmo lltranslate.cmo -c ast_sast.ml
File "ast_sast.ml", line 50, characters 20-24:
Error: This variant expression is expected to have type Sast.sbin_op_fn
       The constructor SFor does not belong to type Sast.sbin_op_fn
Makefile:46: recipe for target 'ast_sast.cmo' failed
make: *** [ast_sast.cmo] Error 2
```

**Comment by Contentmaudlin** [comment]
Yes John changed the SAST and it broke.

Semant PR coming soon which will fix this.

**Comment by Contentmaudlin** [comment]
Addressed in #123
